core/src/main/java/org/elasticsearch/indices/recovery/StartRecoveryRequest.java

```diff
@@ -94,6 +112,11 @@ public void readFrom(StreamInput in) throws IOException {
         targetNode = new DiscoveryNode(in);
         metadataSnapshot = new Store.MetadataSnapshot(in);
         primaryRelocation = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
+            startingSeqNo = in.readZLong();
```

Comment:
it is safe, but I think we should standarize on a single way. We opted for writeLong in other places, so that's why I suggested it here. I'm fine with writeZLong but then we should go to the translog and such and change it there too.